### DISCUSSION TITLE
Execute mdx dataframe leaking memory on server

### DISCUSSION THREAD

**Discussion by raeldor** [cubewise-code/tm1py#941] [General]
Is there anything we need to do after execute mdx dataframe in order to release memory on the server? We are seeing a slow memory leakage that co-incides with our hourly pull of data from the cube. I understand that tm1 holds calculations in memory, but since we are always pulling the same set of data and it hasn’t changed between pulls surely tm1 should be using the same memory that it has already got from the server? Surely it’s not recalculating the data unnecessarily into new memory and keeping the old data in memory never to be used.

**Comment by rclapp** [comment]
TM1py “cleans up” after each query by deleting the cellset as long as you use higher level methods like the dataframe ones. As for what could be causing the memory growth it would be a good idea to turn on performance monitor to see if you can tell what type of “object” is causing the memory to grow. TM1 doesn’t always use garbage memory first, so some memory growth is unavoidable. I would check your VMM settings to make sure they are not too high.

From: Ray Price ***@***.***>
Reply-To: cubewise-code/tm1py ***@***.***>
Date: Friday, July 21, 2023 at 8:59 AM
To: cubewise-code/tm1py ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [cubewise-code/tm1py] Execute mdx dataframe leaking memory on server (Discussion #941)


Is there anything we need to do after execute mdx dataframe in order to release memory on the server? We are seeing a slow memory leakage that co-incides with our hourly pull of data from the cube. I understand that tm1 holds calculations in memory, but since we are always pulling the same set of data and it hasn’t changed between pulls surely tm1 should be using the same memory that it has already got from the server? Surely it’s not recalculating the data unnecessarily into new memory and keeping the old data in memory never to be used.

**Comment by raeldor** [comment]
I appreciate the reply, thank you.  I will try and investigate more using the performance monitor.
